### ISSUE TITLE
No longer able to programmatically access `Microsoft.CodeAnalysis.CSharp.CodeStyle` diagnostics

### ISSUE THREAD

**Issue by tdg5** [dotnet/roslyn#77864]
**Version Used**: 
My trouble started with .NET `9.0.200` but is also present in .NET `9.0.201`

**Steps to Reproduce**:

1.  I created a minimal reproduction as a test case that can be found here: [`tdg5/standard-conventions-cs/Tdg5.StandardConventions.Tests/MinimalReproductionTest.cs`](https://github.com/tdg5/standard-conventions-cs/blob/dotnet-9.0.2-debug/Tdg5.StandardConventions.Tests/MinimalReproductionTest.cs#L42-L80)
2. [This GHA workflow run](https://github.com/tdg5/standard-conventions-cs/actions/runs/14110600692) demonstrates that the test passes for .NET `9.0.1xx` builds, but fails for .NET `9.0.2xx` builds.

**Diagnostic Id**: All diagnostics from `Microsoft.CodeAnalysis.CSharp.CodeStyle` and `Microsoft.CodeAnalysis.CodeStyle`

**Expected Behavior**: I should be able to programmatically access the diagnostics provided by `Microsoft.CodeAnalysis.CSharp.CodeStyle` and `Microsoft.CodeAnalysis.CodeStyle` when a project is loaded via `MSBuildWorkspace.OpenProjectAsync(...)`

**Actual Behavior**: Starting with .NET `9.0.200`, though the `Microsoft.CodeAnalysis.CSharp.CodeStyle` and `Microsoft.CodeAnalysis.CodeStyle` analyzers are detected, they don't appear to expose any `SupportedDiagnostics`.

I first tried creating a discussion, but didn't get any traction. The discussion is here for reference: https://github.com/dotnet/roslyn/discussions/77460

**Comment by tdg5** [comment]
@wtg-mg8 helped me figure out the problem I was hitting in a discussion that can be found here: https://github.com/dotnet/roslyn/discussions/77460#discussioncomment-13774930

Relevant text for future readers:

> Hi @tdg5, we were encountering a similar problem. What was useful for us was adding a handler on the not-so-obvious [AnalyzerFileReference.AnalyzerLoadFailed](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.analyzerfilereference.analyzerloadfailed) event. According to the source code of [GetAnalyzers](https://github.com/dotnet/roslyn/blob/6783c142cac860c27243f666976f556a974a4fb6/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs#L119), this event is fired when errors occur. In our particular case, Microsoft.CodeAnalysis needed to be version 4.12.0, as was being checked [here](https://github.com/dotnet/roslyn/blob/6783c142cac860c27243f666976f556a974a4fb6/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs#L571).

> To handle this error, try breaking out your linq statement from

```csharp
project
        .AnalyzerReferences
        .SelectMany(r => r.GetAnalyzers(project.Language))
        .ToImmutableArray();
```

> to something like:

```csharp
var analyzers = new List<DiagnosticAnalyzer>();
foreach (var analyzerReference in project.AnalyzerReferences)
{
    if (analyzerReference is AnalyzerFileReference analyzerFileReference)
    {
        analyzerFileReference.AnalyzerLoadFailed += (sender, args) =>
        {
            Console.WriteLine($"analyzer reference {analyzerReference.FullPath} failed to load. error code = {args.ErrorCode}, referenced compiled version = {args.ReferencedCompilerVersion}, exception = {args.Exception}");
        };

        var analyzersInReference = analyzerReference.GetAnalyzers(project.Language);
        analyzers.AddRange(analyzersInReference);
    }
}
```

> This should highlight any particular errors that would normally fail silently.

Thanks again @wtg-mg8!
